packages/webcert/src/components/header/WebcertHeaderUnit.test.tsx

```diff
@@ -1,49 +1,81 @@
 import React from 'react'
 import { render, screen } from '@testing-library/react'
-import * as redux from 'react-redux'
-import { Unit, User } from '@frontend/common'
+import { Provider } from 'react-redux'
+import { getChangeUnitResourceLink, getUser, getUserStatistics, getUserWithInactiveUnit } from '@frontend/common'
 import WebcertHeaderUnit from './WebcertHeaderUnit'
+import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
+import dispatchHelperMiddleware, { clearDispatchedActions } from '../../store/test/dispatchHelperMiddleware'
+import apiMiddleware from '../../store/api/apiMiddleware'
+import { userMiddleware } from '../../store/user/userMiddleware'
+import reducer from '@frontend/webcert/src/store/reducers'
+import { updateUser, updateUserResourceLinks, updateUserStatistics } from '../../store/user/userActions'
+import userEvent from '@testing-library/user-event'
 
-const createUserMock = ({ inactiveUnit = false } = {}): User =>
-  ({
-    name: 'Test Testsson',
-    role: 'Läkare',
-    loggedInUnit: { unitId: 'unit', unitName: 'Unit', isInactive: inactiveUnit } as Unit,
-    loggedInCareProvider: { unitName: 'Care provider' } as Unit,
-  } as User)
+let testStore: EnhancedStore
+
+const renderComponent = () => {
+  render(
+    <Provider store={testStore}>
+      <WebcertHeaderUnit />
+    </Provider>
+  )
+}
 
 describe('Webcert header unit', () => {
+  beforeEach(() => {
+    testStore = configureStore({
+      reducer,
+      middleware: (getDefaultMiddleware) => getDefaultMiddleware().prepend(dispatchHelperMiddleware, apiMiddleware, userMiddleware),
+    })
+  })
+
+  afterEach(() => {
+    clearDispatchedActions()
+  })
+
   it('displays care provider and unit that user is logged into', (): void => {
```

Comment:
It's a bit hard to read but it looks like the test to check if the care provider text shows has disappeared